embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/comms/delivery/v2/ExecutionService.kt

```diff
@@ -0,0 +1,8 @@
+package io.embrace.android.embracesdk.internal.comms.delivery.v2
+
+import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
+import java.io.File
+
+interface ExecutionService {
+    fun sendPayload(payload: File): ApiResponse
```

Comment:
Regardless of whether we choose to create an abstraction over `File`, I think it'd be good to be explicit in naming on whether this refers to a JSON payload or a HTTP payload.

Reply:
Right. Definitely a JSON payload, i.e. the thing we stick inside the Body of an HTTP request